packages/odyssey-storybook/src/components/odyssey-mui/Table/Table.stories.tsx

```diff
@@ -0,0 +1,472 @@
+ *
+ * You may obtain a copy of the License at http://www.apache.org/licenses/LICENSE-2.0.
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *
+ * See the License for the specific language governing permissions and limitations under the License.
+ */
+
+import { Story } from "@storybook/react";
+import * as React from "react";
+import Button from "@mui/material/Button";
+import Table from "@mui/material/Table";
+import TableBody from "@mui/material/TableBody";
+import TableCell from "@mui/material/TableCell";
+import TableContainer from "@mui/material/TableContainer";
+import TableHead from "@mui/material/TableHead";
+import TableRow from "@mui/material/TableRow";
+import TableSortLabel from "@mui/material/TableSortLabel";
+import Checkbox from "@mui/material/Checkbox";
+import { MuiThemeDecorator } from "../../../../.storybook/components";
+import TableMdx from "./Table.mdx";
+
+export default {
+  title: `MUI Components/Table`,
+  component: Table,
+  parameters: {
+    docs: {
+      page: TableMdx,
+    },
+  },
+  argTypes: {
+    disabled: {
+      control: "boolean",
+      defaultValue: false,
+    },
+    icon: {
+      control: "text",
+      defaultValue: null,
+    },
+    label: {
+      control: "text",
+      defaultValue: "Asteroids",
+    },
+    wrapped: {
+      control: "boolean",
+      defaultValue: false,
+    },
+  },
+  decorators: [MuiThemeDecorator],
+};
+
+interface Data {
+  name: string;
+  radius: number;
+  type: string;
+  perihelion: string;
+  descriptor: string;
+}
+
+function createData(
+  name: string,
+  radius: number,
+  type: string,
+  perihelion: string,
+  descriptor: string
+): Data {
+  return {
+    name,
+    radius,
+    type,
+    perihelion,
+    descriptor,
+  };
+}
+
+const rows = [
+  createData("Mercury", 2440, "Terrestrial", "--", "Mercurian"),
+  createData("Venus", 6051, "Terrestrial", "--", "Venusian"),
+  createData("Earth", 6378, "Terrestrial", "January 4, 2023", "Terran"),
+  createData("Mars", 3396, "Terrestrial", "June 21, 2022", "Martian"),
+  createData("Jupiter", 71492, "Gas giant", "January 21, 2023", "Jovian"),
+  createData("Saturn", 60268, "Gas giant", "November 29, 2032", "Saturnian"),
+  createData("Uranus", 25559, "Ice giant", "August 17-19, 2050", "Uranian"),
+  createData("Neptune", 24764, "Ice giant", "September 4, 2042", "Neptunian"),
+];
+
+function descendingComparator<T>(a: T, b: T, orderBy: keyof T) {
+  if (b[orderBy] < a[orderBy]) {
+    return -1;
+  }
+  if (b[orderBy] > a[orderBy]) {
+    return 1;
+  }
+  return 0;
+}
+
+type Order = "asc" | "desc";
+
+function getComparator<Key extends keyof any>(
+  order: Order,
+  orderBy: Key
+): (
+  a: { [key in Key]: number | string },
+  b: { [key in Key]: number | string }
+) => number {
```

Comment:
Does this fix the TS warning?

```suggestion
function getComparator<Key extends any>(
  order: Order,
  orderBy: Key
): (
  a: { [key in Key]: number | string },
  b: { [key in Key]: number | string }
) => number {
```

Reply:
Weird, I don't have any warnings in VSC or `typecheck`. Is the GH editor using different rules than we are?

Reply:
Tested locally - this change breaks the typing on `Key`.

Reply:
Played around with it. This fixed it:
```suggestion
function getComparator<Key extends (string | number | symbol)>(
  order: Order,
  orderBy: Key
): (
  a: { [key in Key]: number | string },
  b: { [key in Key]: number | string }
) => number {
```

Playground: https://www.typescriptlang.org/play?jsx=0#code/GYVwdgxgLglg9mABAEwKYGcKrMmYDmAwnALYAOAhgE4VRxUA8AKgHwAUFAXIkwDSIAjbn0T00VAEIBPbgGtUUuMB4BKRAG8AUIkQxlbAQG0xqSVIC6iBogrGq46ebVadOqqiggqSALQBGAG5tRABfYL1EAzsHC0QWG2jTR2dgtw8vJEDgsLTPb0QABiCwzSgpMlREAHl7U0QAXkQAIgpMJsQAH2a0NqDNUEhYBER8D2Jyalp6BgBpBURUAA8obGR0SPQoKjx8TsQwEBIBOq70KSO4ABsVdmCTKm4a8V472rNuOalNFW42YK4NIhDPIpLokJ9zNwDkcTohNtsCKEXjohIDgfM8IgIVDDscqHt4TtQt8GvFoXiNMF3HkkPcGvVGk0ehAmqlEAB+SIUfgCNT1eLM1Y7caUGh0KgcHn8e7SFRs37cwR8+I+QU4YWkUVTCWKgTSt6y4qaIA